sdk/python/feast/diff/FcoDiff.py

```diff
@@ -48,20 +41,28 @@ def add_fco_diff(self, fco_diff: FcoDiff):
         self.fco_diffs.append(fco_diff)
 
 
-Fco = TypeVar("Fco", Entity, BaseFeatureView, FeatureService)
-
-
-def tag_objects_for_keep_delete_add(
+def tag_objects_for_keep_delete_update_add(
     existing_objs: Iterable[Fco], desired_objs: Iterable[Fco]
-) -> Tuple[Set[Fco], Set[Fco], Set[Fco]]:
+) -> Tuple[Set[Fco], Set[Fco], Set[Fco], Set[Fco]]:
     existing_obj_names = {e.name for e in existing_objs}
     desired_obj_names = {e.name for e in desired_objs}
 
     objs_to_add = {e for e in desired_objs if e.name not in existing_obj_names}
-    objs_to_keep = {e for e in desired_objs if e.name in existing_obj_names}
+    objs_to_update = {e for e in desired_objs if e.name in existing_obj_names}
+    objs_to_keep = {e for e in existing_objs if e.name in desired_obj_names}
     objs_to_delete = {e for e in existing_objs if e.name not in desired_obj_names}
 
-    return objs_to_keep, objs_to_delete, objs_to_add
+    return objs_to_keep, objs_to_delete, objs_to_update, objs_to_add
+
+
+FcoProto = TypeVar(
+    "FcoProto",
```

Comment:
Should we maybe just move to `object` instead of `FCO`, since `FCO` isn't a Feast term.

Reply:
am going to do this in a future PR, since it touches a lot of files and don't want to create a massive PR